pkg/apis/config/v1alpha4/default.go

```diff
@@ -79,12 +79,26 @@ func SetDefaultsCluster(obj *Cluster) {
 }
 
 // SetDefaultsNode sets uninitialized fields to their default value.
-func SetDefaultsNode(obj *Node) {
+func SetDefaultsNode(obj *Node, ipFamily ClusterIPFamily) {
 	if obj.Image == "" {
 		obj.Image = defaults.Image
 	}
 
 	if obj.Role == "" {
 		obj.Role = ControlPlaneRole
 	}
+
+	for i := 0; i < len(obj.ExtraPortMappings); i++ {
+		if obj.ExtraPortMappings[i].ListenAddress == "" {
+			if ipFamily == IPv6Family {
+				obj.ExtraPortMappings[i].ListenAddress = defaults.ExtraPortMappingListenAddressIPv6
+			} else {
+				obj.ExtraPortMappings[i].ListenAddress = defaults.ExtraPortMappingListenAddressIPv4
+			}
+		}
+
+		if string(obj.ExtraPortMappings[i].Protocol) == "" {
+			obj.ExtraPortMappings[i].Protocol = PortMappingProtocolTCP
+		}
+	}
```

Comment:
@BenTheElder we just pre-default here, I think is ok, but it will better to have your thoughts

Reply:
I will also remove the defaulting piece from docker/podman providers  if we finalize this approach.

Reply:
I guess this is OK, we can define the behavior based on docker. Not the first time kubernetes has done that.